### PR TITLE
Feature update validation

### PR DESCRIPTION
**Pull request by ammsalme**
* Update record validation to follow rules in https://www.loc.gov/marc/specifications/specrecstruc.html , including:

-  Validate that leader is 24 characters long (if configured with validationOption `leader: true`)
- Validate that leader contains only printable/graphic ASCII characters (if configured with validationOption `characters: true`)

- Validate that field tags are three characters long 
- Validate that field tags consist only of ASCII numeric characters (decimal integers 0-9) and/or ASCII alphabetic characters (uppercase or lowercase, but not both). (if configured with validationOption `characters: true`)
- Validate that controlField tags start with '00' or ASCII alphabetic characters (if configured with validationOption `characters: true`)
- Validate that dataField tags do not start with '00' (if configured with validationOption `characters: true`)

- Validate that controlFields do not have ind1, ind2 or subfields
- Validate that controlField values contain only printable/graphic ASCII characters (if configured with validationOption `characters: true`)
- Validate that controlField value does not exceed maximum length for a field

- Validate that dataFields do not have value

- Validate that indicators are one character long
- Validate that indicators consist only of blanks or ASCII numeric or lowercase alphabetic characters (if configured with validationOption `characters: true`)

- Validate that subfield codes are one character long
- Validate that subfield codes consist only of printable/graphic ASCII characters excluding blank (if configured with validationOption `characters: true`)

- Validate that subfield values do not contain ASCII control characters (if configured with validationOption `noControlCharacters: true`)
- Validate that subfield values do not exceed maximum length for a field

- Validate that fields or subfields do not have additional properties (if configured with validationOption `noAdditionalProperties: true`)

* Add new validationOption `strict: true` that causes all other validationOptions to be handled as `true`

* Add new validationOptions and update default validationOptions:
```
strict: false                  // If true, all validationOptions below are handled as if true

fields: true,                  // Do not allow record without fields
subfields: true,               // Do not allow empty subfields
subfieldValues: true,          // Do not allow subfields without value
controlFieldValues: true       // Do not allow controlFields without value
leader: false,                 // Do not allow record without leader, with empty leader or with leader with length != 24
characters: false              // Do not allow erronous characters in tags, indicators and subfield codes
noControlCharacters: false,    // Do not allow ASCII control characters in field/subfield values
noAdditionalProperties: false  // Do not allow additional properties in fields
```

### CODE REVIEW

File: src/schema.js
Comment (nvolk) [inline src/schema.js:1]: `const indicatorPattern = /^[0-9a-z ]$/u;`

Tähän varmaan /^[0-9 ]$/u ?

File: src/marcFieldSort.js
Comment (nvolk) [inline src/marcFieldSort.js:1]: kauno/fin ja kauno/swe voisi laittaa tulemaan ennen kaunokkia 6XX-kentissä.

Näitä ei tarvitse tähän toteuttaa! Nää on vaan nice-to-have ja allaolevat vähän hankalia.

Tasapelin tullessa, jos molemmissä kentissä on $6 occurrence number, niin isompi voisi voittaa. Samaten $8 \x1.n:t tms olis hauska pakottaa järjestykseen.

Ja sitten olis tietty päheetä, jos vaikka 700:t sorttautusivat $e:n tärkeyden mukaan (säveltäjä > sovittaja > esittäjä).

**Review by nvolk — APPROVED** [review]
Ei mulla mitään ihmeitä tullut vastaan. Oletan hyväksi. Ton varmaan vois korjata:

const indicatorPattern = /^[0-9a-z ]$/u;

Tähän varmaan riittäis /^[0-9 ]$/u ?

Sorttauksen yhteydessä pari epäolennaista huomioita siellä itte filun kohdalla..

File: src/schema.js
Comment (ammsalme) [reply src/schema.js:1]: Nope: myös pienet kirjaimet on ookoo - niitä on tosin harvemmin käytetty.

https://www.loc.gov/marc/specifications/specrecstruc.html#leader

> indicator.
> A data element associated with a data field that supplies additional information about the field. An indicator may
> be any ASCII lowercase alphabetic, numeric, or blank. Indicators are not used in control fields.

File: src/marcFieldSort.js
Comment (ammsalme) [reply src/marcFieldSort.js:1]: Jep - nää olis hyviä.

**Review by natlibfi-jonollil — APPROVED** [review]

**Review by aatuny — APPROVED** [review]

**Pull request merged**